test/ModelTraitTest.php

```diff
@@ -49,23 +51,26 @@ public function testSetData($testData, $expectedProperties)
                 if (!empty($value["objectData"][0])) {
                     foreach ($value["objectData"] as $i => $objData) {
                         foreach ($objData as $subProp => $subValue) {
-                            $this->assertAttributeEquals($subValue, $subProp, $actualData[$i]);
+                            $assertGetter = "get" . ucfirst($subProp);
+                            $this->assertSame($subValue, $actualData[$i]->{$assertGetter}($subProp));
                         }
                     }
                 }
                 continue;
             }
-            $this->assertAttributeEquals($value, $prop, $modelTrait);
+            $getter = "get" . $prop;
```

Comment:
you'd need to run `ucfirst()` on `$prop` here, no?

Reply:
Yep, added. Looks like the tests still worked without it though